tool/planet/agent.go

```diff
@@ -477,6 +503,47 @@ func status(c statusConfig) (ok bool, err error) {
 	return ok, nil
 }
 
+func validateKubernetesService(ctx context.Context, serviceCIDR net.IPNet) error {
+	return utils.Retry(ctx, math.MaxInt64, 5*time.Second, func() error {
+		client, err := cmd.GetKubeClientFromPath(constants.SchedulerConfigPath)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		services := client.CoreV1().Services(metav1.NamespaceDefault)
+		svc, err := services.Get(KubernetesServiceName, metav1.GetOptions{})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if svc.Spec.ClusterIP == "" {
+			// not a clusterIP service?
+			log.Info("kubernetes service clusterIP is empty.")
+			return nil
+		}
+		clusterIP := net.ParseIP(svc.Spec.ClusterIP)
+		if !serviceCIDR.Contains(clusterIP) {
+			log.WithField("cluster-ip", svc.Spec.ClusterIP).
+				Warn("kubernetes service clusterIP is invalid, will recreate service.")
+			return removeKubernetesService(services)
+		}
+		log.Info("kubernetes service clusterIP is valid.")
```

Comment:
Nit: **K**ubernetes

Also, add current CIDR range and cluster IP to the message?